README.md

```diff
@@ -186,14 +166,37 @@ Or:
     }
 ```
 
-* `typeFilter`: (required) A function that is used to identify documents of this type. It accepts as function parameters both (1) the new document and (2) the old document that is being replaced/deleted (if any). For example:
+* `typeFilter`: (required) A function that is used to identify documents of this type. It accepts as function parameters (1) the new document, (2) the old document that is being replaced (if any) and (3) the name of the current document type. For the sake of convenience, a simple type filter function (`simpleTypeFilter`) is available that attempts to match the document's `type` property to the document type's name (e.g. if a document definition is named "message", then a candidate document's `type` property must be "message" to be considered a document of that type). NOTE: In cases where the document is in the process of being deleted, the first parameter's `_deleted` property will be true, so be sure to account for such cases. Also, the second parameter will only be a non-null value if the document is being replaced; it will be null in cases where the old document does not exist or was deleted.
+
+An example of the simple type filter:
 
 ```
-    typeFilter: function(doc, oldDoc) {
+    typeFilter: simpleTypeFilter
+```
+
+And an example of a more complex custom type filter:
+
+```
+    typeFilter: function(doc, oldDoc, currentDocType) {
```

Comment:
This example is now much more complex, but it hopefully provides a more complete picture of what the type filter is capable of.